playwright/pages/components/EnvironmentSelector.ts

```diff
@@ -0,0 +1,61 @@
+import { expect, Locator, Page } from "@playwright/test";
+
+export class EnvironmentSelector {
+  readonly page: Page;
+  readonly loginTextButton: Locator;
+  readonly loginIconButton: Locator;
+  readonly environmentName: Locator;
+  readonly dropdownTrigger: Locator;
+
+  constructor(page: Page) {
+    /**
+     * Components
+     */
+    this.page = page;
+
+    /**
+     * Static locators
+     */
+    this.loginTextButton = page.getByRole("button", {
+      name: "Login required",
+      exact: true,
+    });
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.loginIconButton = page.getByTestId("environment-login-icon-button");
+    this.environmentName = page.getByTestId("active-environment-name");
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.dropdownTrigger = page.getByTestId("environment-dropdown-button");
+  }
+
+  /**
+   * Dynamic locators
+   */
+  getEnvironmentMenuItem(name: string): Locator {
+    return this.page.getByRole("menuitem", { name, exact: true });
+  }
+
+  /**
+   * Actions
+   */
+  async selectEnvironment(name: string) {
+    await this.dropdownTrigger.click();
+    await this.getEnvironmentMenuItem(name).click();
+    await expect(this.environmentName).toHaveText(name);
```

Comment:
⚠️ I think it's better to always check visible elements with:

```
await expect(this.environmentName.getByText(name, { extact: true })).toBeVisible()
```

It ensures it's not hidden some way. WDYT adding something in best practices also?

Reply:
Makes sense to me. 👍 I added a short section in the document.